packages/decorators/src/djs-decorators.ts

```diff
@@ -0,0 +1,121 @@
+import { createFunctionPrecondition, FunctionFallback } from './utils';
+
+export const enum DecoratorIdentifiers {
+	RequiresPermissionsGuildOnly = 'requiresPermissionsGuildOnly',
+	RequiresPermissionsMissingPermissions = 'requiresPermissionsMissingPermissions'
+}
+
+// TODO: Re-enable commented out permissions when updating to DJS v13
+const ServerOnlyPermissions = new Permissions([
+	'BAN_MEMBERS',
+	'CHANGE_NICKNAME',
+	'CONNECT',
+	'CREATE_INSTANT_INVITE',
+	'DEAFEN_MEMBERS',
+	'KICK_MEMBERS',
+	'MANAGE_CHANNELS',
+	'MANAGE_EMOJIS',
+	'MANAGE_GUILD',
+	'MANAGE_MESSAGES',
+	'MANAGE_NICKNAMES',
+	'MANAGE_ROLES',
+	// 'MANAGE_THREADS',
+	'MANAGE_WEBHOOKS',
+	'MENTION_EVERYONE',
+	'MOVE_MEMBERS',
+	'MUTE_MEMBERS',
+	'PRIORITY_SPEAKER',
+	// 'REQUEST_TO_SPEAK',
+	'SEND_TTS_MESSAGES',
+	'SPEAK',
+	'STREAM',
+	// 'USE_APPLICATION_COMMANDS',
+	// 'USE_PRIVATE_THREADS',
+	// 'USE_PUBLIC_THREADS',
+	'USE_VAD',
+	'VIEW_AUDIT_LOG',
+	'VIEW_GUILD_INSIGHTS'
+]);
+
+/**
+ * Allows you to set permissions required for individual methods. This is particularly useful for subcommands that require specific permissions.
+ * @remark This decorator makes the decorated function asynchronous, so any result should be `await`ed.
+ * @param permissionsResolvable Permissions that the method should have.
+ * @example
+ * ```typescript
+ * import { ApplyOptions, RequiresPermissions } from '@sapphire/decorators';
+ * import { SubCommandPluginCommand, SubCommandPluginCommandOptions } from '@sapphire/plugin-subcommands';
+ * import type { Message } from 'discord.js';
+ *
+ * (at)ApplyOptions<SubCommandPluginCommandOptions>({
+ * 	aliases: ['cws'],
+ * 	description: 'A basic command with some subcommands',
+ * 	subCommands: ['add', 'remove', 'reset', { input: 'show', default: true }]
+ * })
+ * export default class extends SubCommandPluginCommand {
+ *     // Anyone should be able to view the result, but not modify
+ * 	public async show(message: Message) {
+ * 		return message.channel.send('Showing!');
+ * 	}
+ *
+ * 	(at)RequiresPermissions('BAN_MEMBERS') // This subcommand should only be available to "staff".
+ * 	public async add(message: Message) {
+ * 		return message.channel.send('Adding!');
+ * 	}
+ *
+ * 	(at)RequiresPermissions('BAN_MEMBERS') // This subcommand should only be available to "staff".
+ * 	public async remove(message: Message) {
+ * 		return message.channel.send('Removing!');
+ * 	}
+ *
+ * 	(at)RequiresPermissions('BAN_MEMBERS') // This subcommand should only be available to "staff".
+ * 	public async reset(message: Message) {
+ * 		return message.channel.send('Resetting!');
+ * 	}
+ * }
+ * ```
+ */
+export const RequiresPermissions = (...permissionsResolvable: PermissionResolvable[]): MethodDecorator => {
+	const resolved = new Permissions(permissionsResolvable);
+	return createFunctionPrecondition((message: Message) => {
+		if (isDMChannel(message.channel) && resolved.any(ServerOnlyPermissions)) {
```

Comment:
Memoize the `resolved.any(ServerOnlyPermissions)` by moving it to the parent function.